File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string answer;
        int count = 0;
        for(auto num:s)
        {
            if(num == '(')
            {
                if(count > 0)
                {
                    answer.push_back(num);
                }
                count++;
            }
            else
            {
                count--;
                if(count > 0)
                {
                    answer.push_back(num);
                }
            }

        }
        return answer;
        
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int n = s.length();

        int count = 0;
        StringBuilder str = new StringBuilder(); 

        for(int i=0;i<n;i++)
        {
            if(s.charAt(i) == '(')
            {
                if(count > 0)
                {
                    str.append(s.charAt(i));
                }
                count++;
            }
            else
            {
                count--;
                if(count > 0)
                {
                    str.append(s.charAt(i));
                }
            }
        }
        return str.toString();
        
        
    }
}